src/components/PopupWithForm.js

```diff
@@ -1,19 +1,33 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, initialButtonText) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     console.log(this._popupForm);
     this._handleFormSubmit = handleFormSubmit;
     console.log(this._closeButton);
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._initialButtonText = initialButtonText;
   }
 
   close() {
     this._popupForm.reset();
     super.close();
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      if (this._initialButtonText === "Save") {
+        this._submitButton.textContent = "Saving...";
+      } else if (this._initialButtonText === "Create") {
+        this._submitButton.textContent = "Creating...";
+      }
+    } else {
+      this._submitButton.textContent = this._initialButtonText;
+    }
+  }
+
   _getInputValues() {
     const inputs = this._popupForm.querySelectorAll(".modal__input");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.